From f5a4e93694bd86c31f1a421b526449cb4898cdb7 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:56:51 +1000
Subject: [PATCH] Avoid unnecessary stat calls

Solus is stateless, so conflicts are already avoided without having to create
new files.

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/core/kconfig.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/core/kconfig.cpp b/src/core/kconfig.cpp
index 6b49874..412617c 100644
--- a/src/core/kconfig.cpp
+++ b/src/core/kconfig.cpp
@@ -659,17 +659,13 @@ QStringList KConfigPrivate::getGlobalFiles() const
     QMutexLocker locker(&s_globalFilesMutex);
     if (s_globalFiles()->isEmpty()) {
         const QStringList paths1 = QStandardPaths::locateAll(QStandardPaths::GenericConfigLocation, QStringLiteral("kdeglobals"));
-        const QStringList paths2 = QStandardPaths::locateAll(QStandardPaths::GenericConfigLocation, QStringLiteral("system.kdeglobals"));
 
         const bool useEtcKderc = !etc_kderc.isEmpty();
-        s_globalFiles()->reserve(paths1.size() + paths2.size() + (useEtcKderc ? 1 : 0));
+        s_globalFiles()->reserve(paths1.size() + (useEtcKderc ? 1 : 0));
 
         Q_FOREACH (const QString &dir1, paths1) {
             s_globalFiles()->push_front(dir1);
         }
-        Q_FOREACH (const QString &dir2, paths2) {
-            s_globalFiles()->push_front(dir2);
-        }
 
         if (useEtcKderc) {
             s_globalFiles()->push_front(etc_kderc);
--
2.16.3
